gapic-generator-java/src/main/java/com/google/api/generator/gapic/composer/common/AbstractServiceClientClassComposer.java

```diff
@@ -802,11 +804,18 @@ private static List<MethodDefinition> createMethodVariants(
             methodVariantBuilder.setReturnType(methodOutputType).setReturnExpr(rpcInvocationExpr);
       }
 
+      List<AnnotationNode> annotations = new ArrayList<>();
       if (method.isDeprecated()) {
-        methodVariantBuilder =
-            methodVariantBuilder.setAnnotations(
-                Arrays.asList(AnnotationNode.withType(TypeNode.DEPRECATED)));
+        annotations.add(AnnotationNode.withType(TypeNode.DEPRECATED));
+      }
+
+      if (method.isInternalApi()) {
+        annotations.add(
+            AnnotationNode.withTypeAndDescription(
+                typeStore.get("InternalApi"), INTERNAL_API_WARNING));
       }
```

Comment:
similar logic is repeated a couple of times within this class. perhaps worth to extract to a helper method.